### PR TITLE
feat: support mimalloc as optional global allocator

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Changes introduced in this pull request:
- Support [`mimalloc`](https://github.com/microsoft/mimalloc#performance) which claims to outperform `jemalloc`
- Note that `rocksdb` does not support `mimalloc` as optional allocator so this feature only benefits `paritydb`

I've run `paritydb` stress test locally, memory usage with `mimalloc` is ~5% lower than with `jemalloc` (~240MB vs ~250MB)
```console
git clone https://github.com/paritytech/parity-db.git
cd parity-db
cargo run -p parity-db-admin --release -- stress
```

## Reference issue to close (if applicable)

Closes 

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

**Review by LesnyRumcajs — APPROVED** [review]

**Review by tyshko5 — APPROVED** [review]

**Review by elmattic — APPROVED** [review]

**Comment by elmattic** [comment]
Did you do some more benchmark? Like some import/validation/export? Peak memory while running a node.

I would be curious to see `VSZ` peak memory too.

**Comment by hanabi1224** [comment]
> Did you do some more benchmarks? Like some import/validation/export? Peak memory while running a node.
> 
> I would be curious to see max `VSZ` metric too (this is tracedk by bench.rb script but not displayed).

@elmattic Not yet, I will update the numbers soon

File: scripts/benchmark_db/bench.rb [hunk 1]
```diff
@@ -115,34 +115,42 @@ def exec_command(command, dry_run)
   metrics
 end
 
+# rubocop:disable Metrics/MethodLength, Metrics/AbcSize
 def write_import_table(metrics)
-  result = "Bench | Import Time | Import RSS | DB Size\n"
-  result += "-|-|-|-\n"
+  result = "Bench | Import Time | Import RSS | Import VSZ | DB Size\n"
```
Comment (hanabi1224) [inline scripts/benchmark_db/bench.rb:120]: @elmattic could u take another look at `bench.rb` changes, thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,8 @@ Notable updates:
 
 ### Added
 * [database] added ParityDb statistics to the stats endpoint. [#2444](https://github.com/ChainSafe/forest/pull/2444)
-
+* [forest-daemon] support jemalloc as optional global allocator [#2477](https://github.com/ChainSafe/forest/pull/2477)
+* [forest-daemon] support mimalloc as optional global allocator [#2506](https://github.com/ChainSafe/forest/pull/2506)
```
Comment (LesnyRumcajs) [inline CHANGELOG.md:7]: Hm, after a bit of thought, I am not sure this should be included in the changelog. It's not really a user-facing feature. It's completely internal to Forest devs - eventually we will most likely pick only one global allocator. Thoughts?

Comment (hanabi1224) [reply CHANGELOG.md:7]: Makes sense, let me remove them

**Review by creativcoder — APPROVED** [review]

**Comment by hanabi1224** [comment]
@elmattic the import time is kinda on par with jemalloc.  I failed with the `bench.rb` script, for some reason it's stuck there without importing the snapshot or crash 

|db|import time|db size|
|--|--|--|
|paritydb+jemalloc|1796s|114GB|
|paritydb+mimalloc|1822s|114GB|

**Pull request merged**